commonmark/src/main/java/org/commonmark/internal/DocumentParser.java

```diff
@@ -481,12 +532,21 @@ private void processInlines() {
      * its parent, and so on until we find a block that can accept children.
      */
     private void addChild(OpenBlockParser openBlockParser) {
-        while (!getActiveBlockParser().canContain(openBlockParser.blockParser.getBlock())) {
-            closeBlockParsers(1);
-        }
+    	// BlankLine nodes do _not_ interrupt the current node type's processor, as
+    	//    they are non-standard CommonMark nodes meant only for roundtrip processing
+    	if(!(openBlockParser.blockParser instanceof BlankLineParser)) {
```

Comment:
I agree this would be good, but I wasn't able to find a way to do it.

Most of the time what is being captured in a `BlankLine` is occurring in `DocumentParser.parseLine(CharSequnce ln)`. That method never seems to deal with individual blocks, instead capturing whole parsers and deciding whether to add their results to the structure via `addChild`. The parsers themselves are also not equipped to contain multiple nodes per parser, as they each contain a single object of the type they're supposed to parse (usually called `block`).

The only place I saw where individual nodes could be added directly was `InlineParserImpl.parseInline()`, which (aside from inline processing being too late for capturing blank lines) still used `Collections.singletonList` and thus was not meant for more than one node at a time either.

Is there a place I missed where we might add the `BlankLine` nodes directly? I agree that it would be cleaner and would likely clear the extra lines out of `addChild`, it just didn't seem to fit within the existing design.